user_guide_src/source/database/results.rst

```diff
@@ -500,6 +515,13 @@ Class Reference
 		Generates an array of ``stdClass`` objects containing
 		field meta-data.
 
+    .. php:method:: getNumRows()
+
+		:returns:	Number of rows in result set
+		:rtype:	int
+
+		Returns number of rows returned by the query
```

Comment:
It would probably be more compact to warn of possible slow performance with SQLite3. I also strongly disagree that it is only beneficial for a getUnbufferedRow situation. For example:
```php
// one fultext query I have takes 7 seconds to run due to unavoidable table scans
// the table scans are also required even if it's just a COUNT(*) query
$query = $db->query("SOME GIANT ELABORATE SQL QUERY PAINFUL TO RUN AND THEN RUN AGAIN AS COUNT QUERY");
if ($query->getNumRows() > 1000) {
  // sometimes instantiating too many records can cause a fatal OUT OF MEMORY ERROR
  echo "You need to make your search more specific.";
}
```

How about this message instead:

> Because `SQLite3` lacks an efficient method returning a record count, CodeIgniter will fetch and buffer the query result records internally and return a count of the resulting record array, which can be inefficient.

Reply:
Okay, the "efficiency" part is the most important here.

Reply:
I've put that changed comment in the docs. I had neglected to indent the subsequent lines of the note. Just submitted a fixed commit.